### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by outscale-jju**

### CODE REVIEW

**Comment by outscale-jju** [comment]
PING;ok;same host;1;2;0.063;0.115;0.251
PING;ok;different host;1;3;0.056;0.069;0.110
TCP;ok;same host;1;2;3255247535
TCP;ok;different host;1;3;4045486008
UDP;ok;same host;1;2;800886968

**Comment by outscale-mgo** [comment]
why did you create a new file "function.sh" instead of refactoring the tests/function.sh ?

File: benchmarks/run_benchmarks.sh [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+            break ;;
+        *)
+            usage; exit 1 ;;
+    esac
+done
+
+# check arguments
+
+if [ -z "$source_dir" ]; then
+    echo "Error: Butterfly source dir not set"
```
Comment (outscale-mgo) [inline benchmarks/run_benchmarks.sh]: have you try an echo -e "...\n" instead of the empty echo ?

File: benchmarks/run_benchmarks.sh [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    echo ""
+    usage
+    exit 1
+fi
+
+if [ -z "$build_dir" ]; then
+    echo "Error: Butterfly build dir not set"
+    echo ""
```
Comment (outscale-mgo) [inline benchmarks/run_benchmarks.sh]: same here

File: benchmarks/run_benchmarks.sh [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    usage
+    exit 1
+fi
+
+if [ -z "$ip_a" ]; then
+    echo "Error: IP of machine A not set"
+    echo ""
```
Comment (outscale-mgo) [inline benchmarks/run_benchmarks.sh]: same

File: benchmarks/run_benchmarks.sh [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    echo "Error: Butterfly source dir seems bad: $f does not exist"
+    echo ""
+    usage
+    exit 1
+fi
+
+f=$build_dir/CMakeCache.txt
+if [ ! -f $f ]; then
+    echo "Error: Butterfly build dir seems bad: $f does not exist"
+    echo ""
```
Comment (outscale-mgo) [inline benchmarks/run_benchmarks.sh]: same

File: benchmarks/run_benchmarks.sh [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+    echo "Error: IP of machine B not set"
+    echo ""
+    usage
+    exit 1
+fi
+
+f=$source_dir/CMakeLists.txt
+if [ ! -f $f ]; then
+    echo "Error: Butterfly source dir seems bad: $f does not exist"
+    echo ""
```
Comment (outscale-mgo) [inline benchmarks/run_benchmarks.sh]: same

File: benchmarks/run_benchmarks.sh [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+    usage
+    exit 1
+fi
+
+case "$package" in
+    rpm) ;;
+    deb) ;;
+    *)
+        echo "Invalid package type $package"
+        echo ""
```
Comment (outscale-mgo) [inline benchmarks/run_benchmarks.sh]: same

File: README.md [hunk 7]
```diff
@@ -140,7 +140,30 @@ Butterfly is based on:
...
+ --------------------+--------------------+--------------------|
+| Ping (min/average) |  0.052ms / 0.118ms |  0.62ms / 0.172ms  |
+| TCP                |     3.1 Gbits/s    |     4.0 Gbits/s    |
+| UDP                |     808 Mbits/s    |     806 Mbits/s    |
+```
+
+Notes:
+- We only use one dedicated core for Butterfly
+- Tests are done without any tweaks on thread scheduling (coming soon)
+- UDP tests are _bad_, XXX gso ?
```
Comment (outscale-mgo) [inline README.md]: gso won't improve udp a lot, for this I suspect a bad scheduling.

File: api/server/graph.cc [hunk 8]
```diff
@@ -109,7 +109,7 @@ bool Graph::start(int argc, char **argv) {
 
     // Create nic brick
     nic = Brick(Pg::nic_new_by_id("port-0", 0), Pg::destroy);
-    if (nic.get() == NULL) {
+    if (nic.get() == NULL || !Pg::nic_set_mtu(nic.get(), 2000)) {
```
Comment (outscale-mgo) [inline api/server/graph.cc]: you can remove `nic.get() == NULL` and this break the test-api

**Pull request closed without merging**